celery/worker/consumer/consumer.py

```diff
@@ -181,7 +181,8 @@ def __init__(self, on_task_request,
                  pool=None, app=None,
                  timer=None, controller=None, hub=None, amqheartbeat=None,
                  worker_options=None, disable_rate_limits=False,
-                 initial_prefetch_count=2, prefetch_multiplier=1, **kwargs):
+                 initial_prefetch_count=2, prefetch_multiplier=1, url=None, **kwargs):
```

Comment:
Add new parameter to allow passing the url directly when instantiating a new Consumer but use None for the default behavior